src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Mvc.Version1_X/MvcViewDocumentClassifierPass.cs

```diff
@@ -39,7 +40,7 @@ protected override void OnDocumentStructureCreated(
             @class.ClassName = CSharpIdentifier.GetClassNameFromPath(filePath);
         }
 
-        @class.BaseType = IntermediateToken.CreateCSharpToken("global::Microsoft.AspNetCore.Mvc.Razor.RazorPage<TModel>");
+        @class.BaseType = new BaseTypeIntermediateNode("global::Microsoft.AspNetCore.Mvc.Razor.RazorPage<TModel>", location: null);
```

Comment:
Should add a constructor that just takes the component parts rather than doing string manipulation in the main ctor in these cases?